test/e2e/httpproxy/httpproxy_test.go

```diff
@@ -193,93 +193,109 @@ var _ = Describe("HTTPProxy", func() {
-					},
-					Spec: certmanagerv1.IssuerSpec{
-						IssuerConfig: certmanagerv1.IssuerConfig{
-							CA: &certmanagerv1.CAIssuer{
-								SecretName: "ca-cert",
-							},
+			// Issuer based on CA to generate new certs with.
+			basedOnCAIssuer := &certmanagerv1.Issuer{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      "ca-issuer",
+				},
+				Spec: certmanagerv1.IssuerSpec{
+					IssuerConfig: certmanagerv1.IssuerConfig{
+						CA: &certmanagerv1.CAIssuer{
+							SecretName: "ca-cert",
 						},
 					},
-				}
-				require.NoError(f.T(), f.Client.Create(context.TODO(), basedOnCAIssuer))
+				},
+			}
+			require.NoError(f.T(), f.Client.Create(context.TODO(), basedOnCAIssuer))
 
-				// Backend client cert, can use for upstream validation as well.
-				backendClientCert := &certmanagerv1.Certificate{
-					ObjectMeta: metav1.ObjectMeta{
-						Namespace: namespace,
-						Name:      "backend-client-cert",
-					},
-					Spec: certmanagerv1.CertificateSpec{
-						Usages: []certmanagerv1.KeyUsage{
-							certmanagerv1.UsageClientAuth,
-						},
-						CommonName: "client",
-						SecretName: "backend-client-cert",
-						IssuerRef: certmanagermetav1.ObjectReference{
-							Name: "ca-issuer",
-						},
+			// Backend client cert, can use for upstream validation as well.
+			backendClientCert := &certmanagerv1.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      "backend-client-cert",
+				},
+				Spec: certmanagerv1.CertificateSpec{
+					Usages: []certmanagerv1.KeyUsage{
+						certmanagerv1.UsageClientAuth,
 					},
-				}
-				require.NoError(f.T(), f.Client.Create(context.TODO(), backendClientCert))
-
-				contourConfig.TLS = config.TLSParameters{
-					ClientCertificate: config.NamespacedName{
-						Namespace: namespace,
-						Name:      "backend-client-cert",
+					CommonName: "client",
+					SecretName: "backend-client-cert",
+					IssuerRef: certmanagermetav1.ObjectReference{
+						Name: "ca-issuer",
 					},
-				}
+				},
+			}
+			require.NoError(f.T(), f.Client.Create(context.TODO(), backendClientCert))
 
-				contourConfiguration.Spec.Envoy.ClientCertificate = &contour_api_v1alpha1.NamespacedName{
-					Name:      "backend-client-cert",
+			contourConfig.TLS = config.TLSParameters{
+				ClientCertificate: config.NamespacedName{
 					Namespace: namespace,
-				}
-			})
+					Name:      "backend-client-cert",
+				},
+			}
 
+			contourConfiguration.Spec.Envoy.ClientCertificate = &contour_api_v1alpha1.NamespacedName{
+				Name:      "backend-client-cert",
+				Namespace: namespace,
+			}
+		})
+		Context("with backend tls", func() {
 			testBackendTLS(namespace)
 		})
+		Context("with backend tls version configured via Contour ConfigMap", func() {
+			protocolVersion := "TLSv1.3"
+
+			Context("via Contour ConfigMap", func() {
+				BeforeEach(func() {
+					contourConfig.Cluster.UpstreamTLS.MinimumProtocolVersion = protocolVersion
```

Comment:
seems like this should this be `1.3` rather than `TLSv1.3` as above? (though echoserver might return `TLSv1.3`, i don't remember exactly)

Reply:
yeah, I had a feeling that might not align.

Based on [this block](https://github.com/kubernetes-sigs/ingress-controller-conformance/blob/d920ed36a0076e169a9a329a850844ab3a695ae8/images/echoserver/echoserver.go#L231C6-L246) that @tsaarni shared with me, I'm pretty sure echoserver returns "TLSv1.3".

I don't think that would also be acceptable for configuring Contour though

Reply:
I just wanted to commit something and see all the tests go (my machine suffers when I try to run tests locally lol)